src/iosrv/obj.c

```diff
@@ -71,41 +184,41 @@ ds_obj_retry_cb(tse_task_t *task, void *arg)
 		return rc;
 	}
 
-	/* Register the callback again, because it will be removed
-	 * after callback.
+	/*
+	 * Register the retry callback again, because it has been removed
+	 * from the completion callback list. If this registration failed,
+	 * the task will just stop retry on next run.
 	 */
-	rc = dc_task_reg_comp_cb(task, ds_obj_retry_cb, arg, sizeof(arg));
+	rc = dc_task_reg_comp_cb(task, dsc_obj_retry_cb, arg, sizeof(*oh));
```

Comment:
It is more clear:
rc = dc_task_reg_comp_cb(task, dsc_obj_retry_cb, oh, sizeof(*oh));